internal/caddy/config.go

```diff
@@ -22,58 +21,47 @@ type Storage struct {
 type Config struct {
 	Storage Storage                `json:"storage"`
 	Apps    map[string]interface{} `json:"apps"`
+	Logging caddy.Logging          `json:"logging"`
 }
 
 // ControllerConfig represents ingress controller config received through cli arguments.
 type ControllerConfig struct {
-	Email          string
-	AutomaticTLS   bool
-	TLSUseStaging  bool
 	WatchNamespace string
+	ConfigMapName  string
 }
 
-// NewConfig returns a plain slate caddy2 config file.
-func NewConfig(namespace string, cfg ControllerConfig) *Config {
-	acmeIssuer := caddytls.ACMEIssuer{
-		CA:    getCAEndpoint(cfg.TLSUseStaging),
-		Email: cfg.Email}
+// NewConfig returns a base plain slate caddy2 config file.
+func NewConfig(namespace string, cfgMapConfig *Config) *Config {
+	var cfg *Config
 
-	return &Config{
-		Storage: Storage{
-			System: "secret_store",
-			StorageValues: StorageValues{
-				Namespace: namespace,
-			},
-		},
-		Apps: map[string]interface{}{
-			"tls": caddytls.TLS{
-				Automation: &caddytls.AutomationConfig{
-					Policies: []*caddytls.AutomationPolicy{
-						{
-							IssuerRaw: caddyconfig.JSONModuleObject(acmeIssuer, "module", "acme", nil),
-						},
-					},
+	if cfgMapConfig != nil {
+		cfg = cfgMapConfig
+	} else {
+		cfg = &Config{
+			Logging: caddy.Logging{},
+			Apps: map[string]interface{}{
+				"tls": &caddytls.TLS{
+					CertificatesRaw: caddy.ModuleMap{},
 				},
-				CertificatesRaw: caddy.ModuleMap{},
-			},
-			"http": caddyhttp.App{
-				Servers: map[string]*caddyhttp.Server{
-					"ingress_server": &caddyhttp.Server{
-						AutoHTTPS: &caddyhttp.AutoHTTPSConfig{
-							Disabled: !cfg.AutomaticTLS,
-							Skip:     make([]string, 0),
+				"http": &caddyhttp.App{
+					Servers: map[string]*caddyhttp.Server{
+						"ingress_server": {
+							AutoHTTPS: &caddyhttp.AutoHTTPSConfig{},
+							Listen:    []string{":443"},
```

Comment:
We remove the port 80 from the Listen is that correct?

Reply:
Yes, to enable HTTP->HTTPS by default. Looking at it again made me realize that it may be an mistake 🤔 I don't know if users will still be able to connect from an incompatible hostname (say the IP address of the node).

@mholt I couldn't find docs on rules about Automatic HTTP->HTTPS redirects and Listen ports. If we listen to 443 with Automatic HTTP->HTTPS redirects, is it possible to connect with HTTP only?

Reply:
@Embraser01 Unless [explicitly disabled](https://caddyserver.com/docs/json/apps/http/servers/#automatic_https/disable), automatic HTTPS is always enabled for all listeners other than the HTTP port. If a server listens on ports 80 and 443, for example, HTTPS will only be enabled on port 443. You don't need to listen on port 80 if you just want redirects to HTTPS (recommended).

Reply:
@mholt @Embraser01 I need to better investigate it but if we listen only on port 443 who redirect the client from port 80 to 443? Let suppose we have:

Client -> LoadBalancer -> [CaddyIngressController && Ingress && Caddy] -> Backend service


If the client ask for http://example.caddy/test ok port 80 at the moment the request go trough the LoadBalancer and reach the CaddyConfig that is listening only on port 443, so when we LB try to connect to port 80 of caddy I suppose will receive "unable to connect", I expect that the LB can connect and caddy will return a 301 (or something similar) to redirect the client to httpS://example.caddy/test , I'm missing something?

PS: I not tested so just speculating, I'll try to test ASAP.